core/src/main/java/org/mskcc/oncotree/model/TumorType.java

```diff
@@ -28,6 +28,56 @@
     private Integer level = null;
     private static final String UNKNOWN_ONCOTREE_NODE_LEVEL = "-1";
 
+    public TumorType() {}
+
+    public TumorType(TumorType otherTumorType) {
+        this.id = otherTumorType.id;
+        this.code = otherTumorType.code;
+        this.color = otherTumorType.color;
+        this.name = otherTumorType.name;
+        if (otherTumorType.mainType != null) {
+            this.mainType = new MainType(otherTumorType.mainType);
+        }
+        this.NCI = new ArrayList<String>(otherTumorType.NCI);
+        this.UMLS = new ArrayList<String>(otherTumorType.UMLS);
+        this.tissue = otherTumorType.tissue;
+        // shallow copy
+        this.children = new HashMap<String, TumorType>(otherTumorType.children);
+        this.parent = otherTumorType.parent;
+        this.deprecated = otherTumorType.deprecated;
+        this.history = new ArrayList<String>(otherTumorType.history);
+        if (otherTumorType.links != null) {
+            this.links = new Links(otherTumorType.links);
+        }
+        this.level = otherTumorType.level;
+    }
+
+    public TumorType deepCopy() {
+        TumorType newTumorType = new TumorType();
```

Comment:
maybe a helper function private void shallowCopyAllButChildren(TumorType otherTumorType) would be helpful and allow reuse?